lib/src/util/posix/dynamic_library.cc

```diff
@@ -41,7 +41,11 @@ namespace facter { namespace util {
         close();
 
         // Don't actually perform a load to determine if it is already loaded
+#ifdef _AIX
+        _handle = dlopen(name.c_str(), RTLD_LAZY);
```

Comment:
I believe this was @branan's workaround for FACT-891, and without this, Facter won't compile on AIX. He mentioned earlier that he was debating whether to leave this hack in until we fix FACT-891, so we don't push a commit that doesn't actually build in AIX.